File: poh9/poh9_8.rb
#!/usr/bin/ruby
path=gets
a=path.split.join('/').split('/')
st=[]
a.each{|e|
	if e=='.' || e==''
	elsif e=='..'
		if !st.empty? && st[-1]!='..'
			st.pop
		else
			#warn('We have Directory Traversal')
			st<<'..' if path[0]!='/'
	else
		st<<e
	end
}
r=(path[0]=='/' ? '/' : '')+st*'/'
puts r.empty? ? '.' : r

__END__
puts File.expand_path gets.split.join('/')

File: poh9/poh9_8.py
#!/usr/bin/python
def simplify_path(path):
	a=path.split('/')
	st=[]
	for e in a:
		if e=='.' or e=='':
			pass
		elif e=='..':
			if st and st[-1]!='..':
				st.pop()
			else:
				#warn('We have Directory Traversal')
				if path[0]!='/': st.append('..')
		else:
			st.append(e)
	r=('/' if path[0]=='/' else '')+'/'.join(st)
	return r if r else '.'

if __name__=='__main__':
	if True:
		print(simplify_path('/'.join(input().split())))
